jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -560,9 +580,45 @@ static Chunk asChunk(ByteBuffer byteBuffer, boolean last, Retainable retainable)
          * @param failure the cause of the failure
          * @return a new Error.Chunk
          */
-        static Error from(Throwable failure)
+        static Chunk from(Throwable failure)
+        {
+            return from(failure, true);
+        }
+
+        /**
+         * <p>Creates an {@link Error error chunk} with the given failure.</p>
+         *
+         * @param failure the cause of the failure
+         * @param last true if the failure is terminal, else false for transient failure
+         * @return a new Error.Chunk
+         */
+        static Chunk from(Throwable failure, boolean last)
         {
-            return new Error(failure);
+            return new Chunk()
+            {
+                public Throwable getError()
+                {
+                    return failure;
+                }
+
+                @Override
+                public ByteBuffer getByteBuffer()
+                {
+                    return BufferUtil.EMPTY_BUFFER;
+                }
+
+                @Override
+                public boolean isLast()
+                {
+                    return last;
+                }
+
+                @Override
+                public String toString()
+                {
+                    return String.format("%s@%x{c=%s,l=%b}", getClass().getSimpleName(), hashCode(), failure, last);
```

Comment:
The generated class name will be confusing. How about hardcoding `ErrorContent@%x{...` instead of using `"%s@%x{...", getClassGetSimpleName()`?

Reply:
Every time I do that, @sbordet changes it back to `getSimpleName()`!   Why is it confusing?  If it is, then the extended class can always implement toString itself.

Reply:
I find it confusing in this particular case because it is in an anonymous class, so the logs will look like `Chunk$1@12345{...}`.